server/server.go

```diff
@@ -222,47 +222,46 @@ func (s *Server) handleGetMh(w lookupResponseWriter, r *http.Request) {
 	log := logger.With("multihash", mh)
 	ctx := context.Background()
 
+	// create result and error channels
 	resChan := make(chan model.ProviderResult)
-	errChan := make(chan error)
+	// the size of two is required so that the client doesn't in its go routine when trying to submit an error and return
+	errChan := make(chan error, 2)
 
+	// launch the find in a separate go routine
 	go s.c.FindAsync(ctx, mh, resChan, errChan)
```

Comment:
I really think that `DHashClient.FindAsync` needs to change. I think it would be better if it did not take an error channel, since it only returns one error. Consider calling it like this:
```go
go func() {
    // FindAsync returns results on resChan until there are no more results or error.
    // When finished, returns the error or nil.
    err = s.c.FindAsync(ctx, mh, resChan)
}()
for res := range resChan {
    // process result
}
// FindAsync finished, check for error.
if err != nil {
    s.handleError(w, err, log)
}
```

Reply:
Looks great - lets do that!